sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java

```diff
@@ -712,11 +712,18 @@ private void addExposedTypes(Class<?> clazz, Class<?> cause) {
   private void addExposedTypes(Invokable<?, ?> invokable, Class<?> cause) {
     addExposedTypes(invokable.getReturnType(), cause);
     for (Annotation annotation : invokable.getAnnotations()) {
+      Class<? extends Annotation> annotationClass = annotation.annotationType();
+      if (!Modifier.isPublic(annotationClass.getModifiers())) {
+        // Non-public annotations are invisible to Beam users. Example: Guava's
+        // ParametricNullness is package-private and exists to support Kotlin platform
```

Comment:
Interesting! Definitely relevant beyond Kotlin (for example https://checkerframework.org/releases/0.9.7/doc/checkers/nullness/quals/PolyNull.html) but your change is more general and obviously correct.

Reply:
Thank you for reference. I updated the comment:

> This annotation helps static analyzers including Kotlin platform.